lib/SILOptimizer/Transforms/SILMem2Reg.cpp

```diff
@@ -512,6 +597,11 @@ void MemoryToRegisters::removeSingleBlockAllocation(AllocStackInst *ASI) {
     // value.
     if (auto *SI = dyn_cast<StoreInst>(Inst)) {
       if (SI->getDest() == ASI) {
+        if (SI->getOwnershipQualifier() == StoreOwnershipQualifier::Assign) {
```

Comment:
`store x [init] y` is also consuming, right? Shouldn't this check be `ownership != Unqualified | Trivial`?

Reply:
Ah, I see. Don't you need to destroy `SI->getSrc()` too?

Reply:
Since [init] in a way 'forwards' the ownership to the stack location, we don't need to destroy the src. We'll have a later destroy_addr of the stack location which gets transformed separately.

Reply:
But I think the `destroy_addr` will just get removed, right? I think we either need to add a `destroy_value` here or make sure we transform the `destory_addr` into a `destroy_value`.

Also what about stores with the `Assign` ownership qualifier? The source of those instructions also needs to get destroyed, right?

Reply:
https://github.com/apple/swift/blob/0b456ba795f6e9aa25e1910a8bcad901567943c6/lib/SILOptimizer/Transforms/SILMem2Reg.cpp#L376 The pass already does transform the `destroy_addr` to the `destroy_value` of the `RunningVal`.
Same for [assign]

Reply:
Ah, I missed that. OK, I'm satisfied 🙃